service/controller/v11/drainer_resource_set.go

```diff
@@ -39,7 +43,27 @@ func NewDrainerResourceSet(config DrainerResourceSetConfig) (*controller.Resourc
 		}
 	}
 
+	var endpointResource controller.Resource
+	{
+		c := endpoint.Config{
+			G8sClient: config.G8sClient,
+			K8sClient: config.K8sClient,
+			Logger:    config.Logger,
+		}
+
+		ops, err := endpoint.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		endpointResource, err = toCRUDResource(config.Logger, ops)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
 	resources := []controller.Resource{
+		endpointResource,
```

Comment:
To make this work more nicely I had to get the endpoint resource from endpoint operator in. This makes the PR more ugly but the outcome is really cool. We start to retire the endpoint-operator and get finalizer support for the endpoint magic which makes it finally stable. We also have more control of the whole process of draining and such because this has to work hand in hand with the rest of the update procedure.